Sources/SwiftParser/Parser+Entry.swift.gyb

```diff
@@ -0,0 +1,75 @@
+%{
+  from gyb_syntax_support import *
+  # -*- mode: Swift -*-
+  # Ignore the following admonition it applies to the resulting .swift file only
+}%
+//// Automatically Generated From Entry.swift.gyb.
+//// Do Not Edit Directly!
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_spi(RawSyntax) import SwiftSyntax
+
+extension Parser {
+  /// Parse the source code in the given string as Swift source file. See
+  /// `Parser.init` for more details.
+  public static func parse(
+    source: String,
+    parseTransition: IncrementalParseTransition? = nil
+  ) -> SourceFileSyntax {
+    var parser = Parser(source)
+    return SourceFileSyntax.parse(from: &parser)
+  }
+
+  /// Parse the source code in the given string as Swift source file. See
+  /// `Parser.init` for more details.
+  public static func parse(
+    source: UnsafeBufferPointer<UInt8>,
+    maximumNestingLevel: Int? = nil,
+    parseTransition: IncrementalParseTransition? = nil
+  ) -> SourceFileSyntax {
+    var parser = Parser(source, maximumNestingLevel: maximumNestingLevel)
+    return SourceFileSyntax.parse(from: &parser)
+  }
+}
+
+public protocol SyntaxParseable: SyntaxProtocol {
+  static func parse(from parser: inout Parser) -> Self
+}
+
+% for node in SYNTAX_NODES:
+%   if node.parser_function:
+extension ${node.name}: SyntaxParseable {
+  public static func parse(from parser: inout Parser) -> Self {
+    let node = parser.${node.parser_function}()
+    return parser.parseRemainder(into: node).syntax
+  }
+}
+
+%   end
+% end
+fileprivate extension Parser {
+  mutating func parseRemainder<R: RawSyntaxNodeProtocol>(into: R) -> R {
+    guard let layout = into.raw.layoutView else {
+      return into
+    }
+
+    let remainingTokens = self.consumeRemainingTokens()
+    if remainingTokens.isEmpty {
+      return into
+    }
+
+    let unexpected = RawUnexpectedNodesSyntax(elements: remainingTokens, arena: self.arena)
+    let withUnexpected = layout.replacingChild(at: layout.children.count - 1, with: unexpected.raw, arena: self.arena)
```

Comment:
This breaks if the node is a syntax collection because syntax collections are not interleaved with unexpected nodes. Could you use the syntax node structure (I don't recall exactly what I ended up calling it, it was one of my latest PRs) to assert that the node we are parsing is a layout node and has unexpected nodes as the last child?

Reply:
Sure, if I can find it :). FWIW these are only generated for layout nodes, I think I used to actually specifically check that in the gyb but then removed it since no non-layout nodes have parse functions anyway.